dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -381,41 +388,38 @@ protected void githubWebhookRelease(String repository, String username, String g
-            // '&=' does not short-circuit, ensuring that all of the lists are processed.
-            // 'isSuccessful &= x()' is equivalent to 'isSuccessful = isSuccessful & x()'.
+            // The intermediate lambda events from processing each entry type are combined into one lambda event
             List<Service12> services = dockstoreYaml12.getService() != null ? List.of(dockstoreYaml12.getService()) : List.of();
-            isSuccessful &= createWorkflowsAndVersionsFromDockstoreYml(services, repository, gitReference, installationId, username, dockstoreYml, Service.class, messageWriter);
-            isSuccessful &= createWorkflowsAndVersionsFromDockstoreYml(dockstoreYaml12.getWorkflows(), repository, gitReference, installationId, username, dockstoreYml, BioWorkflow.class, messageWriter);
-            isSuccessful &= createWorkflowsAndVersionsFromDockstoreYml(dockstoreYaml12.getTools(), repository, gitReference, installationId, username, dockstoreYml, AppTool.class, messageWriter);
-            isSuccessful &= createWorkflowsAndVersionsFromDockstoreYml(dockstoreYaml12.getNotebooks(), repository, gitReference, installationId, username, dockstoreYml, Notebook.class, messageWriter);
+            lambdaEvent.combine(createWorkflowsAndVersionsFromDockstoreYml(services, repository, gitReference, installationId, username, dockstoreYml, Service.class, messageWriter));
+            lambdaEvent.combine(createWorkflowsAndVersionsFromDockstoreYml(dockstoreYaml12.getWorkflows(), repository, gitReference, installationId, username, dockstoreYml, BioWorkflow.class, messageWriter));
+            lambdaEvent.combine(createWorkflowsAndVersionsFromDockstoreYml(dockstoreYaml12.getTools(), repository, gitReference, installationId, username, dockstoreYml, AppTool.class, messageWriter));
+            lambdaEvent.combine(createWorkflowsAndVersionsFromDockstoreYml(dockstoreYaml12.getNotebooks(), repository, gitReference, installationId, username, dockstoreYml, Notebook.class, messageWriter));
 
         } catch (Exception ex) {
 
             // If an exception propagates to here, log something helpful and abort .dockstore.yml processing.
-            isSuccessful = false;
-            String msg = "User " + username + ": Error handling push event for repository " + repository + " and reference " + gitReference + "\n" + generateMessageFromException(ex);
+            lambdaEvent.setSuccess(false);
+            String msg = "User " + username + ": Error handling push event for repository " + repository + " and reference " + gitReference + "\n- " + generateMessageFromException(ex);
             LOG.info(msg, ex);
             messageWriter.println(msg);
             messageWriter.println("Terminated processing of .dockstore.yml.");
 
-            throw new CustomWebApplicationException(msg, statusCodeForLambda(ex));
-
+            if (throwIfNotSuccessful) {
+                throw new CustomWebApplicationException(msg, statusCodeForLambda(ex));
+            }
```

Comment:
This is the fix to prevent the GitHub installation from throwing an exception. We forgot to check the `throwIfNotSuccessful` boolean before throwing

Reply:
Good find!